core/domain/stats_domain_test.py

```diff
@@ -1316,3 +1318,342 @@ def test_calculation_output_must_be_less_than_one_million_bytes(self):
+        learner_answer_details = stats_domain.LearnerAnswerDetails.from_dict(
+            learner_answer_details_dict)
+        self.assertEqual(
+            learner_answer_details.state_reference, 'exp_id.state_name')
+        self.assertEqual(
+            learner_answer_details.entity_type, 'exploration')
+        self.assertEqual(
+            learner_answer_details.interaction_id, 'TextInput')
+        self.assertEqual(
+            len(learner_answer_details.learner_answer_info_list), 1)
+        self.assertEqual(
+            learner_answer_details.learner_answer_info_list[0].answer,
+            'This is my answer')
+        self.assertEqual(
+            learner_answer_details.learner_answer_info_list[0].answer_details,
+            'This is my answer details')
+        self.assertEqual(
+            learner_answer_details.learner_answer_info_list[0].created_on,
+            datetime.datetime(2019, 6, 19, 13, 59, 29, 153073))
+        self.assertEqual(
+            learner_answer_details.accumulated_answer_info_json_size_bytes,
+            4000)
+        self.assertEqual(
+            learner_answer_details.schema_version, 1)
+
+    def test_add_learner_answer_info(self):
+        learner_answer_info = stats_domain.LearnerAnswerInfo(
+            'id_2', 'This answer', 'This details',
+            datetime.datetime.strptime('27 Sep 2012', '%d %b %Y'))
+        self.assertEqual(
+            len(self.learner_answer_details.learner_answer_info_list), 1)
+        self.learner_answer_details.add_learner_answer_info(
+            learner_answer_info)
+        self.assertEqual(
+            len(self.learner_answer_details.learner_answer_info_list), 2)
+
+    def test_delete_learner_answer_info(self):
+        self.assertEqual(
+            len(self.learner_answer_details.learner_answer_info_list), 1)
+        learner_answer_info = stats_domain.LearnerAnswerInfo(
+            'id_2', 'This answer', 'This details',
+            datetime.datetime.strptime('27 Sep 2012', '%d %b %Y'))
+        self.learner_answer_details.add_learner_answer_info(
+            learner_answer_info)
+        self.assertEqual(
+            len(self.learner_answer_details.learner_answer_info_list), 2)
+        self.learner_answer_details.delete_learner_answer_info('id_1')
+        self.assertEqual(
+            len(self.learner_answer_details.learner_answer_info_list), 1)
+        self.assertNotEqual(
+            self.learner_answer_details.accumulated_answer_info_json_size_bytes,
+            0)
+
+    def test_update_state_reference(self):
+        self.assertEqual(
+            self.learner_answer_details.state_reference, 'exp_id.state_name')
+        self.learner_answer_details.update_state_reference(
+            'exp_id_1.state_name_1')
+        self.assertEqual(
+            self.learner_answer_details.state_reference,
+            'exp_id_1.state_name_1')
+
+
+class LearnerAnswerDetailsValidationTests(test_utils.GenericTestBase):
```

Comment:
Usually all tests for a class (in this case LearnerAnswerDetails) are in the same test class. So, I'd suggest renaming the class above to LearnerAnswerDetailsTests and moving all this into that.

Reply:
Done